recipes/tre/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.apple import fix_apple_shared_install_name
+from conan.tools.build import cross_building
+from conan.tools.cmake import cmake_layout
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv
+from conan.tools.files import copy, get, replace_in_file, rm, rmdir
+from conan.tools.gnu import Autotools, AutotoolsToolchain
+from conan.tools.microsoft import is_msvc
+
+required_conan_version = ">=1.53.0"
+
+
+class TreConan(ConanFile):
+    name = "tre"
+    description = "TRE is a lightweight, robust, and efficient POSIX-compliant regexp matching library with some exciting features such as approximate (fuzzy) matching."
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/laurikari/tre"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = "regex", "fuzzy matching"
+
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration("Windows builds are not supported yet")
+
+    def build_requirements(self):
+        self.tool_requires("libtool/2.4.7")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+        tc = AutotoolsToolchain(self)
+        tc.generate()
+
+    def _patch_sources(self):
+        if self.settings.os == "Windows" and not is_msvc(self):
+            replace_in_file(self, os.path.join(self.source_folder, "win32", "tre.def"), "tre.dll", "libtre.dll")
```

Comment:
So this can't actually happen right now right? The validate() method above removes any Windows configurations. Do you expect then that once it works this would be needed?

Reply:
I would like to add Windows support for both this and the related libmagic recipe. tre needs some minor fixes in Windows, but I would like to merge the simpler version wihout it first, so the dependent packages can at least be developed further.
This bit of logic is a carry-over from the Vcpkg port I based this recipe on.